File: ui/base/ime/events.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "ui/base/ime/events.h"

namespace ui {
namespace {

// If this property is present, then it means the input field should suppress
// key autorepeat.
constexpr char kPropertySuppressAutoRepeat[] = "suppress_auto_repeat";

}  // namespace

bool HasKeyEventSuppressAutoRepeat(const ui::Event::Properties& properties) {
  return properties.count(kPropertySuppressAutoRepeat);
}

void SetKeyEventSuppressAutoRepeat(ui::Event::Properties& properties) {
  properties[kPropertySuppressAutoRepeat] = {};
}

}  // namespace ui

File: chrome/browser/resources/webui_browser/tab_strip/events.ts
// Copyright 2026 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import type {Tab as TabData} from '/tab_strip_api/tab_strip_api_data_model.mojom-webui.js';
import type {NodeId} from '/tab_strip_api/tab_strip_api_types.mojom-webui.js';

export type TabClosed = NodeId;

export type TabActivated = TabData;

export type TabUpdated = TabData;

export interface TabAdded {
  id: NodeId;
  isActive: boolean;
}
